overlays/nodejs/lambda/lambdaMixins.ts

```diff
@@ -0,0 +1,302 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as crypto from "crypto";
+
+import * as arn from "../arn";
+import * as iam from "../iam";
+import * as runtime from "./runtimes";
+import * as utils from "../utils";
+
+import * as lambdaFunction from "./function";
+import * as permission from "./permission";
+
+/**
+ * Context is the shape of the context object passed to a Function callback.
+ */
+export interface Context {
+    callbackWaitsForEmptyEventLoop: boolean;
+    readonly functionName: string;
+    readonly functionVersion: string;
+    readonly invokedFunctionArn: string;
+    readonly memoryLimitInMB: string;
+    readonly awsRequestId: string;
+    readonly logGroupName: string;
+    readonly logStreamName: string;
+    readonly identity: any;
+    readonly clientContext: any;
+    getRemainingTimeInMillis(): string;
+}
+
+/**
+ * EntryPoint is the signature for a serverless function that will be invoked each time the AWS
+ * Lambda is invoked.
+ *
+ * This function can be synchronous or asynchronous function, though async is only supported with an
+ * AWS Lambda runtime of 8.10 or higher.  On those runtimes a Promise can be returned, 'callback'
+ * parameter can be ignored, and AWS will appropriately handle things. For AWS lambda pre-8.10, a
+ * synchronous function must be provided.  The synchronous function should return nothing, and
+ * should instead invoke 'callback' when complete.
+ */
+export type EntryPoint<E, R> = (event: E, context: Context, callback: (error: any, result: R) => void) => Promise<R> | void;
+
+/**
+ * EntryPointFactory is the signature for a function that will be called once to produce the
+ * serverless function that AWS Lambda will invoke.  It can be used to initialize expensive state
+ * once that can then be used across all invocations of the Lambda (as long as the Lambda is using
+ * the same warm node instance).
+ */
+export type EntryPointFactory<E, R> = () => EntryPoint<E, R>;
+
+/**
+ * An EventHandler is either a JavaScript Function instance or an aws.lambda.Function that can be
+ * used to handle an event triggered by some resource.  If just a JavaScript function is provided
+ * the AWS Lambda will be created by calling [createFunction] on it.  If more control over the
+ * resultant AWS Lambda is required, clients can call [createFunction] directly and pass the result
+ * of that to any code that needs an EventHandler.
+ */
+export type EventHandler<E, R> = EntryPoint<E, R> | lambdaFunction.Function;
+
+
+export interface CodePathOptions {
```

Comment:
Nit, type-level comment and maybe properties too?

Reply:
This was removed.  it was temporary until i updated this PR to reference hte latest @pulumi/pulumi.